doc/development.md

```diff
@@ -0,0 +1,10 @@
+# Development of Nixpkgs {#part-development}
```

Comment:
This is a new manual section intended to contain development information relevant to _users_, such as:

- How to report bugs or request a change (using the issue tracker)
- How to test pull requests (without reviewing the code)
- How to check if merged pull requests are available in releases

But the section isn't there yet, for now it just contains a starting blurb and then some moved documentation from before.